src/components/SearchParamsContext/SearchParamsContext.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { URLSearchParamsInit, useSearchParams } from 'react-router-dom';
+
+interface ContextValue {
+  searchParams: URLSearchParams,
+  setSearchParams: (
+    nextInit: URLSearchParamsInit,
+    navigateOptions?: {
+      replace?: boolean | undefined;
+      state?: any,
+    } | undefined) => void
+}
+
+export const SearchParamsContext = React.createContext<ContextValue>({
```

Comment:
I have also been thinking about adding context, but I realized that `useSearchParams` gives us the ability to interact with `Search Params` anywhere in our program. So `Search Params` are global by nature, it doesn't make much sense to create a context for them, in my opinion

Reply:
@sergey-lipikhin a good explanation, I agree with you.